config/configmapprovider/provider.go

```diff
@@ -40,15 +45,27 @@ type Provider interface {
 	// contains the value to be injected in the configuration and the corresponding watcher that
 	// will be used to monitor for updates of the retrieved value.
 	//
-	// watcher callback is called when the config changes. watcher may be called from
+	// `selector` must follow the "<scheme>:<opaque_data>" format. This format is compatible with the URI definition
```

Comment:
It is not clear why this is called `selector`. Would `location` or `url` be a better name?

Reply:
It is not really an URL, I had that version but it doesn't fit well for env. @Aneurysm9 had a comment that in order to have env:NAME you had to look for host in the URL and he did not like it.

Location may be better, will update

Reply:
Done.